File: frontend/src/types/Flow.ts
import type { Step } from './Step'

export interface Flow {
  id: string
  name: string
  description?: string
  created_at: string
  steps: Step[]
}

export interface FlowCreate {
  name: string
  description?: string
}

export interface FlowUpdate {
  name: string
  description?: string
}

File: backend/app/schemas/flow.py
from pydantic import BaseModel
import uuid
from datetime import datetime
from typing import List, Optional

from .step import Step

class FlowBase(BaseModel):
    name: str
    description: Optional[str] = None

class FlowCreate(FlowBase):
    pass

class FlowUpdate(FlowBase):
    pass

class FlowInDBBase(FlowBase):
    id: uuid.UUID
    created_at: datetime

    class Config:
        from_attributes = True

class Flow(FlowInDBBase):
    steps: List[Step] = []

class FlowInDB(FlowInDBBase):
    pass
